Support/AnyLayer.swift

```diff
@@ -0,0 +1,230 @@
+import TensorFlow
+import _Differentiation
+
+/// A helper that stops the program with an error when an erased derivative type does not
+/// match up with the true underlying type.
+@inline(never)
+@usableFromInline
+internal func derivativeTypeMismatch(
+  _ x: Any.Type, _ y: Any.Type, file: StaticString = #file, line: UInt = #line
```

Comment:
Consider:

```suggestion
  got: Any.Type, expected: Any.Type, file: StaticString = #file, line: UInt = #line
```

(And of course the corresponding improvement to the resulting error message.)